### ISSUE TITLE
Keyboard navigations issues in new settings menu

### ISSUE THREAD

**Issue by Nardol** [home-assistant/frontend#12594]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Before Home Assistant 2022.5.0, in Configuration it was possible to navigate using Tab and Shift+Tab and select what we want to configure, which is not the case now.
When going to Settings and press tab, focus go to Home Assistant Cloud which is OK but pressing Tab again leave the page so it is not possible to navigate between settings menu items.
Moreover, pressing up/down does not help as focus move alone so we have to be very quick to be able to select what we want before focus get placed to the beginning of the menu, which make impossible a first discover of the new menu for example.

### Describe the behavior you expected

Pressing Tab and Shift+Tab should allow us to navigate to something else than Home Assistant Cloud, we should be able to navigate in all the settings and if possible, up and down arrow in browse mode should not move the focus, in fact focus should not be changed alone

### Steps to reproduce the issue

1. Go to Settings
2. navigate in the settings menu using Tab and Shift+Tab

### What version of Home Assistant Core has the issue?

2022.5.0

### What was the last working version of Home Assistant Core?

2022.4.7 

### In which browser are you experiencing the issue with?

Firefox 100 and Google Chrome 101.0.4951.54

### Which operating system are you using to run this browser?

Debian Buster 10.12

### Additional information

The focus moving alone when using up and down arrow in browse mode (so the screen reader mode to use specifics navigation keys and read text using up and down arrow) might be the same problem I observe in #8414

**Comment by steverep** [comment]
This menu has had bad ARIA for a long time but it was still usable.  Now it looks like the ARIA is even worse and keyboard event handlers were introduced, which I don't understand the reasoning behind at all.

**Comment by steverep** [comment]
@Nardol this menu is now fully trying to use the `mwc-list` component, which has horrible accessibility.  Check out the [demo page](https://material-components.github.io/material-web/demos/list/).

@zsarnett it looks like you committed these changes.  Help me understand a couple things:
1. It looks like the whole point of `ha-clickable-list-item` is to inject an anchor element wrapper.  Why go through that trouble instead of just using a click event on the `mwc-list-item`?
2. What is the reason behind blurring the `ha-clickable-list-item` element on click?

I'm sure this won't fix the problems, but right off the bat I see the anchor has `aria-role="option"`...
1. `aria-role` is not an attribute.  It's just `role`.
2. The `option` role is not valid here anyway as it is not a descendant of a `listbox` or `group`.

Fixing this will be painful without patching `mwc-list`.

**Comment by zsarnett** [comment]
We can look into the issue and try and fix this asap.

The reason for the `ha-clickable-list-item` was to take advantage of our router and not refresh the page each time.

@bramkragten may have better answers here but I will talk with him and see what we can do

**Comment by steverep** [comment]
@zsarnett  the reason I ask is because wrapping/nesting nesting `<a>` elements around/within other elements which have other interactive roles (e.g. button, option, tab, etc.) causes confusion for the browser and assistive technology.  This is common throughout the frontend and should be fixed.  I'll open another issue for that.

In any case, the anchor is not the main problem for this issue - `mwc-list` is just very dumb about setting ARIA roles properly.  I can examine the code further and open an issue with them too.

**Comment by steverep** [comment]
@Nardol the PR I submitted would not make it navigable using using tab, but rather it will act like a typical menu, i.e. tab enters the menu and triggers focus mode and then up/down/home/end navigates the menu items.

**Comment by Nardol** [comment]
@steverep I've just tried #12845 and navigation is really better in settings.
Tested with Firefox 101 on Debian Buster.
I have not tried with the Android app.
Thanks!
